src/modules/Game.class.js

```diff
@@ -1,68 +1,393 @@
+
+    this.boardState[rowIdx][colIdx] = randomValue;
+
+    this.updateBoard();
+  }
+
+  addTouchEvents() {
+    this.gameBoard.addEventListener('touchstart', (e) => {
+      const touchStart = e.touches[0];
+
+      this.touchStartX = touchStart.pageX;
+      this.touchStartY = touchStart.pageY;
+    });
+
+    this.gameBoard.addEventListener('touchend', (e) => {
+      const touchEnd = e.changedTouches[0];
+      const deltaX = touchEnd.pageX - this.touchStartX;
+      const deltaY = touchEnd.pageY - this.touchStartY;
+
+      if (Math.abs(deltaX) > Math.abs(deltaY)) {
+        if (deltaX > 0) {
+          this.moveRight();
+        } else {
+          this.moveLeft();
+        }
+      } else {
+        if (deltaY > 0) {
+          this.moveDown();
+        } else {
+          this.moveUp();
+        }
+      }
+
+      this.generate();
+      this.result();
+    });
+  }
+
+  handleKeydown(userEvent) {
+    switch (userEvent.key) {
+      case 'ArrowUp':
+        this.moveUp();
+        break;
+      case 'ArrowDown':
+        this.moveDown();
+        break;
+      case 'ArrowLeft':
+        this.moveLeft();
+        break;
+      case 'ArrowRight':
+        this.moveRight();
+        break;
+    }
+
+    this.result();
+  }
+
+  getStatus() {
+    if (this.boardState.some((row) => row.includes(2048))) {
+      this.status = 'won';
+    } else if (this.boardState.some((row) => row.includes(0))) {
+      this.status = 'playing';
+    } else {
+      for (let i = 0; i < 4; i++) {
+        for (let j = 0; j < 4; j++) {
+          const cell = this.boardState[i][j];
+
+          if (
+            (i > 0 && cell === this.boardState[i - 1][j]) ||
+            (i < 3 && cell === this.boardState[i + 1][j]) ||
+            (j > 0 && cell === this.boardState[i][j - 1]) ||
+            (j < 3 && cell === this.boardState[i][j + 1])
+          ) {
+            this.status = 'playing';
+
+            return 'playing';
+          }
+        }
+      }
+      this.status = 'lost';
+    }
+
+    this.result();
+
+    return this.status;
+  }
+
+  result() {
+    this.messageLose.classList.add('hidden');
+    this.messageWin.classList.add('hidden');
+
+    if (this.status === 'lost') {
+      this.messageLose.classList.remove('hidden');
+    } else if (this.status === 'won') {
+      this.messageWin.classList.remove('hidden');
+    }
+  }
+
+  moveLeft() {
+    if (!this.gameStarted) {
+      return;
+    }
+
+    for (let rowIdx = 0; rowIdx < 4; rowIdx++) {
+      const newRow = this.boardState[rowIdx].filter((val) => val !== 0);
+      const mergedRow = [];
+
+      for (let i = 0; i < newRow.length; i++) {
+        if (newRow[i] === newRow[i + 1]) {
+          mergedRow.push(newRow[i] * 2);
+          this.score += newRow[i] * 2;
+          i++;
+        } else {
+          mergedRow.push(newRow[i]);
+        }
+      }
+
+      while (mergedRow.length < 4) {
+        mergedRow.push(0);
+      }
+
+      this.boardState[rowIdx] = mergedRow;
+    }
+
```

Comment:
This logic is similar in every method, try to
put it in a helper function and use it everywhere